flink-clients/src/main/java/org/apache/flink/client/CliFrontend.java

```diff
@@ -860,8 +910,50 @@ protected Properties getYarnProperties() throws IOException {
 		return yarnProperties;
 	}
 	
-	protected Client getClient(CommandLine line, ClassLoader classLoader) throws IOException {
-		return new Client(getJobManagerAddress(line), getGlobalConfiguration(), classLoader);
+	protected Client getClient(CommandLine line, ClassLoader classLoader, String programName) throws IOException {
+		String jmAddrString = getJobManagerAddressString(line);
+		InetSocketAddress jobManagerAddress = null;
+		if(jmAddrString.equals(YARN_DEPLOY_JOBMANAGER)) {
+			System.out.println("YARN cluster mode detected. Switching Log4j output to console");
+			LogManager.getRootLogger().addAppender(new ConsoleAppender(new PatternLayout(DEFAULT_LOG4J_PATTERN_LAYOUT)));
```

Comment:
Do we have to hardwire log4j into our code? That contradicts slf4j's purpose. Would be great if we could get rid of that.